providers/slurm/slurm.go

```diff
@@ -195,6 +195,13 @@ func (p *Provider) UpdatePod(ctx context.Context, pod *v1.Pod) error {
 // DeletePod deletes pod.
 func (p *Provider) DeletePod(ctx context.Context, pod *v1.Pod) error {
 	log.Printf("Delete %s", podName(pod.Namespace, pod.Name))
+	pi := p.pods[podName(pod.Namespace, pod.Name)]
+	if pi.jobID != 0 {
+		_, err := p.slurmAPI.CancelJob(context.Background(), &sAPI.CancelJobRequest{JobId: pi.jobID})
+		if err != nil {
+			return errors.Wrapf(err, "can't cancel job %d", pi.jobID)
```

Comment:
should pod bee removed from map either way in this case?

Reply:
No. In such case we can have jobs that will stay in a queue forever. That error can signalize that smth happened with red-box. So better wait till k8s retries to delete pod once again